cg/services/orders/order_summary_service/order_summary_service.py

```diff
@@ -20,27 +19,46 @@ def get_summary(self, order_id: int) -> OrderSummary:
 
     def get_summaries(self, order_ids: list[int]) -> list[OrderSummary]:
         orders: list[Order] = self.store.get_orders_by_ids(order_ids)
-        cases: list[CaseSummary] = self._get_case_summaries(order_ids)
         analyses: list[AnalysisSummary] = self.analysis_client.get_summaries(order_ids)
-        return create_summaries(orders=orders, analysis_summaries=analyses, case_summaries=cases)
+        order_summaries = self.create_summaries(orders=orders, analysis_summaries=analyses)
+        return order_summaries
 
-    def _get_case_summaries(self, order_ids: list[int]):
-        summaries: list[CaseSummary] = []
+    def create_summaries(
+        self, orders: list[Order], analysis_summaries: list[AnalysisSummary]
+    ) -> list[OrderSummary]:
+        order_summaries: list[OrderSummary] = []
+        analysis_summary_map: dict = _get_analysis_map(analysis_summaries)
+        for order in orders:
+            analysis_summary: AnalysisSummary = analysis_summary_map.get(order.id)
+            order_summary = self.create_order_summary(order=order, summary=analysis_summary)
+            order_summaries.append(order_summary)
+        return order_summaries
 
-        for order_id in order_ids:
-            not_received: int = self.store.get_case_not_received_count(order_id)
-            in_preparation: int = self.store.get_case_in_preparation_count(order_id)
-            in_sequencing: int = self.store.get_case_in_sequencing_count(order_id)
-            order: Order = self.store.get_order_by_id(order_id)
-            failed_sequencing_qc: int = get_cases_failed_sequencing_qc_count(order)
-
-            summary = CaseSummary(
-                order_id=order_id,
-                not_received=not_received,
-                in_preparation=in_preparation,
-                in_sequencing=in_sequencing,
-                failed_sequencing_qc=failed_sequencing_qc,
-            )
-            summaries.append(summary)
-
-        return summaries
+    def create_order_summary(self, order: Order, summary: AnalysisSummary) -> OrderSummary:
+        order_id = order.id
+        counted_cases: list[str] = summary.case_ids if summary else []
```

Comment:
These are the cases_to_exclude @ChrOertlin. In the AnalysisSummary object you will find an order id and a summary of all the statuses found in Trailblazer. Since the conclusion of the refinement was that these are not to be duplicated, they are excluded when calculating outside-of-tb-statuses.

Reply:
ah and then there is a priority list and something cannot be counted twice in a different list. I see.

Reply:
Essentially that is the deal. But the only priority this PR adds is that if we get a status from TB for a case, then that is the only status for the case.